client/state/data-layer/wpcom/timezones/index.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * External dependencies
+ */
+import mapValues from 'lodash/mapValues';
+import map from 'lodash/map';
+import fromPairs from 'lodash/fromPairs';
+
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+
+import { TIMEZONES_REQUEST } from 'state/action-types';
+
+import {
+	timezonesRequestSuccess,
+	timezonesRequestFailure,
+	timezonesReceive,
+} from 'state/timezones/actions';
+
+/**
+ * Converts an array with the shape [ { value: 'foo', label: 'bar' } ]
+ * to a keyed object { foo: 'bar' }.
```

Comment:
thanks for the visual in the comment!

fun fact! JSDoc provides for a way to provide examples in our code

```js
/**
 * @typedef {{value: String, label: String}} ValueLabelRecord
 */

/**
 * @typedef {Object.<String, String>} ValueLabelMap
 */

/**
 * Converts an value/label pairs from API into object whose
 * keys are the values and whose values are the labels.
 *
 * @example
 * valueLabelToObject( [ { value: 'foo', label: 'bar' }, { value: 'biz', label: 'bat' } ] )
 * // returns { foo: 'bar', biz: 'bat' }
 *
 * @param {ValueLabelRecord[]} pairs timezone values and display labels
 * @returns {ValueLabelMap} object whose keys are timezone values, values are timezone labels
```

Further, if you think that the lodash utilities here are more readable then please keep them. Personally I would find this possibly clearer with a reduction, so I'll post some code just for an alternative approach:

```js
const timezonePairsToMap = pairs =>
	pairs.reduce( ( timezoneMap, ( { value, label } ) ) => ( {
		...timezoneMap,
		[ value ]: label,
	} ), {} );
```